File: Java/InsertionSort/BJAlgo_2750.java
import java.io.*;

public class BJAlgo_2750 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] dataList = new int[N];
        for (int i = 0; i < N; i++) {
            dataList[i] = Integer.parseInt(br.readLine());
        }

        for (int stand = 0; stand < dataList.length - 1; stand++) {
            for (int num = stand + 1; num > 0; num--) {
                if (dataList[num] < dataList[num - 1]) {
                    int tmp = dataList[num];
                    dataList[num] = dataList[num - 1];
                    dataList[num - 1] = tmp;
                }
            }
        }

        for (int i = 0; i < dataList.length; i++) {
            System.out.println(dataList[i]);
        }
    }
}

File: Python/InsertionSort/BJAlgo_2750.py
from sys import stdin

N = int(stdin.readline())
data_list = list()
for _ in range(N):
    data_list.append(int(stdin.readline()))

for stand in range(len(data_list) - 1):
    for num in range(stand + 1, 0, -1):
        if data_list[num] < data_list[num - 1]:
            data_list[num], data_list[num - 1] = data_list[num - 1], data_list[num]
        else:
            break

for i in range(len(data_list)):
    print(data_list[i])